core/services/fluxmonitor/flux_monitor.go

```diff
@@ -378,284 +328,210 @@ func NewPollingDeviationChecker(
-
-		waitOnStop: make(chan struct{}),
+		pollDelay:      pollDelay,
+		chMaybeLogs:    make(chan maybeLog),
+		chStop:         make(chan struct{}),
+		waitOnStop:     make(chan struct{}),
 	}, nil
 }
 
 // Start begins the CSP consumer in a single goroutine to
 // poll the price adapters and listen to NewRound events.
-func (p *PollingDeviationChecker) Start(ctx context.Context) error {
+func (p *PollingDeviationChecker) Start() {
 	logger.Debugw("Starting checker for job",
 		"job", p.initr.JobSpecID.String(),
 		"initr", p.initr.ID)
 
-	err := p.fetchAggregatorData()
-	if err != nil {
-		return err
-	}
-
-	ctx, p.cancel = context.WithCancel(ctx)
-
-	roundSubscription, err := p.fluxAggregator.SubscribeToLogs(ctx, nil)
-	if err != nil {
-		return err
-	}
-
-	_, err = p.poll(p.threshold)
-	if err != nil {
-		return err
-	}
-
-	go p.consume(ctx, roundSubscription)
-	return nil
+	go p.consume()
 }
 
 // Stop stops this instance from polling, cleaning up resources.
 func (p *PollingDeviationChecker) Stop() {
-	if p.cancel != nil {
-		p.cancel()
-		<-p.waitOnStop
-	}
+	close(p.chStop)
+	<-p.waitOnStop
 }
 
-// stopTimer stops and clears the timer as suggested by the documentation.
-func stopTimer(arg *time.Timer) {
-	if !arg.Stop() && len(arg.C) > 0 {
-		// Residual events are the timer's channel and need to be cleared.
-		//
-		// Refer to timer.Stop's documentation or
-		// https://developpaper.com/detailed-explanation-of-the-trap-of-timer-in-golang/
-		<-arg.C
+func (p *PollingDeviationChecker) OnConnect() {
+	p.connected.Set(true)
+}
+
+func (p *PollingDeviationChecker) OnDisconnect() {
+	p.connected.Set(false)
+}
+
+func (p *PollingDeviationChecker) HandleLog(log interface{}, err error) {
+	select {
+	case p.chMaybeLogs <- maybeLog{log, err}:
+	case <-p.chStop:
 	}
 }
 
-func (p *PollingDeviationChecker) consume(ctx context.Context, roundSubscription contracts.LogSubscription) {
-	defer roundSubscription.Unsubscribe()
+func (p *PollingDeviationChecker) consume() {
+	defer close(p.waitOnStop)
+
+	unsubscribeLogs := p.fluxAggregator.SubscribeToLogs(p)
+	defer unsubscribeLogs()
 
-	idleThreshold := p.idleThreshold
-	if idleThreshold == 0 {
-		idleThreshold = math.MaxInt64
+	p.pollIfEligible(p.threshold)
+
+	var idleTimeout <-chan time.Time
+	if p.idleThreshold > 0 {
+		idleTimeout = time.After(p.idleThreshold)
 	}
 
-	idleThresholdTimer := time.NewTimer(idleThreshold)
-	defer stopTimer(idleThresholdTimer)
+	pollTimeout := time.NewTicker(p.pollDelay)
+	defer pollTimeout.Stop()
 
 Loop:
 	for {
-		jobRunTriggered := false
-
 		select {
-		case <-ctx.Done():
-			close(p.waitOnStop)
+		case <-p.chStop:
 			return
-		case maybeLog := <-roundSubscription.Logs():
-			if maybeLog.Error != nil {
-				logger.Error(errors.WithStack(maybeLog.Error))
+		case maybeLog := <-p.chMaybeLogs:
+			if maybeLog.Err != nil {
+				logger.Error(errors.WithStack(maybeLog.Err))
+				// @@TODO: other error handling?
+				// @@TODO: detect broken subscription?
```

Comment:
Actually, subscriptions are handled in LogSubscriber now, so errors here would be from the DecodingLogListener, if anything.